app/src/main/java/de/westnordost/streetcomplete/quests/localized_name/AddLocalizedNameAdapter.kt

```diff
@@ -137,8 +137,7 @@ class AddLocalizedNameAdapter(
         if (languageTag == "international") return context.getString(R.string.quest_streetName_menuItem_international)
         val languageCode = languageTag.substringBefore('-')
         val isRomanization = languageTag.endsWith("Latn")
-        // if Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP was, could use Locale.forLanguageTag
-        val locale = Locale(languageCode)
+        val locale = Locale.forLanguageTag(languageCode)
 
```

Comment:
I think this is wrong, or at least, incomplete. Look at 
https://docs.oracle.com/javase/8/docs/api/java/util/Locale.html#forLanguageTag-java.lang.String-

It takes a language tag like for example `si-Latn`. The information whether something is a romanization (or another transliteration into another script) can be accessed from `locale.script` and `locale.getDisplayScript()`. These functions are only available since LOLLIPOP.
Furthermore, with LOLLIPOP, the huge if-else can be simplified, because we no longer need `R.string.quest_streetName_menuItem_romanized` etc but can just use `locale.getDisplayScript()`

Reply:
Thanks, I'm not very familiar with the `Locale` class.